pkg/socket/types.go

```diff
@@ -0,0 +1,30 @@
+package socket
+
+import (
+	"crypto/tls"
+	"time"
+)
+
+type Socket interface {
+	Connect() error
+	Disconnect() error
+	Command(command string, process Processor) error
+}
+
+type Processor func([]byte) bool
+
+type Config struct {
+	Network Network
+	Address string
+	Timeout time.Duration
+	TlsConf *tls.Config
+}
+
+type Network string
+
+const (
+	NetworkIP   Network = "tcp"
+	NetworkTCP  Network = "tcp"
+	NetworkUDP  Network = "udp"
+	NetworkUnix Network = "unix"
+)
```

Comment:
That is not used anywhere. Why do we need it?

Reply:
This was my mistake. I use them in the tests now. The idea is to give to the implementer the available options. Something similar to what the `http` packages does with the status codes `http.StatusAccepted` etc so that they know what is supported rather than passing strings directly. The is also a bug `NetworkIP   Network = "tcp"` which became `NetworkIP   Network = "ip" but I might do some work here in order to support plain ip transactions.

Reply:
hmm, the thing that I forgot is that IP packets need raw sockets and we will get permission errors so I removed the `IP` option for the time beiing.